pipelines/templates/task-aqua-install.yml

```
@@ -9,15 +9,35 @@ parameters:
   - name: aquaInstallTags
     type: string
     default: '--tags ci'
+  - name: AQUA_GITHUB_TOKEN
+    type: string
+  # Required (loaded from templates/vars.common.yml)
+  - name: AQUA_ROOT_DIR
+    type: string
+  - name: AQUA_BINARY_FULL_INSTALL_PATH
+    type: string
+  # To allow using the aqua installed/other installed version over default agent version
+  - name: GOROOT
+    type: string
 steps:
-  - bash: |
-      curl -sSfL https://raw.githubusercontent.com/aquaproj/aqua-installer/v1.1.2/aqua-installer | bash -s -- -i "$(Agent.ToolsDirectory)/aqua"
-      export AQUA_PACKAGE_BIN="${AQUA_ROOT_DIR:-${XDG_DATA_HOME:-$HOME/.local/share}/aquaproj-aqua}/bin"
-      echo "##vso[task.prependpath]${AQUA_PACKAGE_BIN}"
-      echo "Added ${AQUA_PACKAGE_BIN} to PATH"
+  - pwsh: |
+      $binarydirectory = '${{ parameters.AQUA_BINARY_FULL_INSTALL_PATH }}' | Split-Path -Parent
+      New-Item -ItemType Directory -Path $binarydirectory -ErrorAction SilentlyContinue && Write-Host "✅ Created the directory to house the aqua binary: [$binarydirectory]"
+      New-Item -ItemType Directory -Path '${{ parameters.AQUA_ROOT_DIR }}' -Force -ErrorAction SilentlyContinue && Write-Host "✅ Created the directory to house the aqua binary: [${{ parameters.AQUA_ROOT_DIR }}]"
+      # go run github.com/aquaproj/aqua-installer@latest -o "${{ parameters.AQUA_BINARY_FULL_INSTALL_PATH }}"
+      &curl -sSfL https://raw.githubusercontent.com/aquaproj/aqua-installer/v1.1.2/aqua-installer | bash -s -- -i "${{ parameters.AQUA_BINARY_FULL_INSTALL_PATH }}"
+      Write-Host "##vso[task.prependpath]${{ parameters.AQUA_ROOT_DIR }}/bin"
+      Write-Host "##vso[task.prependpath]$binarydirectory"
     displayName: install-aqua
-    condition: ne(variables['Agent.OS'], 'Windows_NT')
-  - bash: |
-      $(Agent.ToolsDirectory)/aqua install ${{ parameters.aquaInstallMethod }} ${{ parameters.aquaInstallTags }}
+    failOnStderr: false
+    env:
+      AQUA_GITHUB_TOKEN: ${{ parameters.AQUA_GITHUB_TOKEN }}
+      GOROOT: ${{ parameters.GOROOT }}
```

```suggestion
      GOROOT: ${{ parameters.GOROOT }}
```

Shouldn't need on install stage